pkg/ui/src/views/reports/containers/enqueueRange/index.tsx

```diff
@@ -0,0 +1,238 @@
+import _ from "lodash";
+import React from "react";
+import Helmet from "react-helmet";
+import { withRouter, WithRouterProps } from "react-router";
+import { connect } from "react-redux";
+
+import { enqueueRange } from "src/util/api";
+import { cockroach } from "src/js/protos";
+import Print from "src/views/reports/containers/range/print";
+import "./index.styl";
+
+import EnqueueRangeRequest = cockroach.server.serverpb.EnqueueRangeRequest;
+import EnqueueRangeResponse = cockroach.server.serverpb.EnqueueRangeResponse;
+
+const QUEUES = [
+  "replicate",
+  "gc",
+  "merge",
+  "split",
+  "replicaGC",
+  "raftlog",
+  "raftsnapshot",
+  "consistencyChecker",
+  "timeSeriesMaintenance",
+];
+
+interface EnqueueRangeProps {
+  handleEnqueueRange: (queue: string, rangeID: number, nodeID: number, skipShouldQueue: boolean) => Promise<EnqueueRangeResponse>;
+}
+
+interface EnqueueRangeState {
+  queue: string;
+  rangeID: string;
+  nodeID: string;
+  skipShouldQueue: boolean;
+  response: EnqueueRangeResponse;
+  error: Error;
+}
+
+class EnqueueRange extends React.Component<EnqueueRangeProps & WithRouterProps, EnqueueRangeState> {
+  constructor(props: EnqueueRangeProps & WithRouterProps) {
+    super(props);
+    this.state = {
+      queue: QUEUES[0],
+      rangeID: "",
+      nodeID: "",
+      skipShouldQueue: false,
+      response: null,
+      error: null,
+    };
+  }
+
+  handleUpdateQueue = (evt: React.FormEvent<{ value: string }>) => {
+    this.setState({
+      queue: evt.currentTarget.value,
+    });
+  }
+
+  handleUpdateRangeID = (evt: React.FormEvent<{ value: string }>) => {
+    this.setState({
+      rangeID: evt.currentTarget.value,
+    });
+  }
+
+  handleUpdateNodeID = (evt: React.FormEvent<{ value: string }>) => {
+    this.setState({
+      nodeID: evt.currentTarget.value,
+    });
+  }
+
+  handleSubmit = (evt: React.FormEvent<any>) => {
+    evt.preventDefault();
+
+    this.props.handleEnqueueRange(
+      this.state.queue,
+      // These parseInts should succeed because <input type="number" />
+      // enforces numeric input. Otherwise they're NaN.
```

Comment:
It seems like it would be cleaner to explicitly check for those NaNs and issue something like `this.setState({ error: "Range ID must be numeric" })` rather than propagating the erroneous data.

Modern browsers probably do enforce numeric inputs, but we shouldn't trust that.